front/src/app/todo-list/todo-list.component.ts

```diff
@@ -54,6 +63,33 @@ export class TodoListComponent implements OnInit, OnDestroy {
     this.destroy$.complete();
   }
 
+  showNewTaskGroupFormField(): void {
+    this.isInputVisible = false;
+    this.changeDetectorRef.detectChanges()
+    this.newTaskGroupFormField?.nativeElement.focus();
+  }
+
+  hideNewTaskGroupFormField(): void {
+    this.isInputVisible = true;
+  }
+
+  createNewTaskGroup(name: string): void {
+    if (name != '') {

```

Comment:
`if (name)` would've been enough here

But it's event better to decrease the amount of text inside `if` by following restructuring:
```
this.isInputVisible = true;
this.inputText = '';

if (!name) {
  return;
}

this.isResultLoading = true;
this.apiService.createTaskGroup(name)
// rest of the code
```

It's a common trick to avoid nested code and simplify functions